integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java

```diff
@@ -421,8 +434,46 @@ public Row cleanseAndValidateRow(Row row) {
         // Record the results in the appended columns, move processing partition value last
         newValues[schema.length + 1] = newValues[schema.length - 1];
         newValues[schema.length] = sbRejectReason;
-        newValues[schema.length - 1] = (valid ? "1" : "0");
-        return RowFactory.create(newValues);
+        newValues[schema.length - 1] = (rowValid ? "1" : "0");
+
+        CleansedRowResult cleansedRowResult = new CleansedRowResult();
+        cleansedRowResult.Row = RowFactory.create(newValues);
+        cleansedRowResult.ColumnsValid = columnsValid;
+        cleansedRowResult.RowIsValid = rowValid;
+        return cleansedRowResult;
+    }
+
+    // Maps partitions of CleansedRowResults to a count of invalid columns, total valid, total invalid
+    private Iterable<long[]> mapToPartitionValidationCounts(Iterator<CleansedRowResult> cleansedRowResultIterator){
+
+        long[] validationCounts = new long[schema.length + 2];
+
+        while (cleansedRowResultIterator.hasNext()) {
+
+            CleansedRowResult cleansedRowResult = cleansedRowResultIterator.next();
+
+            for (int idx = 0; idx < schema.length; idx++) {
+
+                if (!cleansedRowResult.ColumnsValid[idx])
+                {
+                    validationCounts[idx] = validationCounts[idx] + 1l;
```

Comment:
Not required but best practices suggests using a capital L since a lowercase l can sometimes look like a 1.